scipy/linalg/_basic.py

```diff
@@ -141,11 +145,37 @@ def solve(a, b, lower=False, overwrite_a=False,
     array([ True,  True,  True], dtype=bool)
 
     """
+    xp = array_namespace(a, b)
+    if check_finite:
+        a = _asarray(a, check_finite=True, xp=xp)
+        b = _asarray(b, check_finite=True, xp=xp)
+    if is_numpy(xp):
+        return _solve(a, b, lower=lower, overwrite_a=overwrite_a,
+                      overwrite_b=overwrite_b, assume_a=assume_a, transposed=transposed)
+    unsupported_args = {
+        'lower': lower,
+        'assume_a': assume_a != 'gen',
+        'transposed': transposed
```

Comment:
Are we silently ignoring `overwrite_a` and `overwrite_b` since they don't guarantee anything about the behavior?

Reply:
Yep.